tests/unit/server/api/v1/test_vectors_settings.py

```diff
@@ -27,6 +28,98 @@
 
 @pytest.mark.asyncio
 class TestSuiteVectorsSettings:
+    @pytest.mark.parametrize("role", [UserRole.owner, UserRole.admin])
+    async def test_update_vector_settings(self, async_client: "AsyncClient", role: UserRole):
+        vector_settings = await VectorSettingsFactory.create()
+        user = await UserFactory.create(role=role, workspaces=[vector_settings.dataset.workspace])
+
+        response = await async_client.patch(
+            f"/api/v1/vectors-settings/{vector_settings.id}",
+            headers={API_KEY_HEADER_NAME: user.api_key},
+            json={"title": "New Title"},
+        )
+
+        assert response.status_code == 200
+        assert response.json() == {
+            "id": str(vector_settings.id),
+            "name": vector_settings.name,
+            "title": "New Title",
+            "dimensions": vector_settings.dimensions,
+            "dataset_id": str(vector_settings.dataset_id),
+            "inserted_at": vector_settings.inserted_at.isoformat(),
+            "updated_at": vector_settings.updated_at.isoformat(),
+        }
+
+        assert vector_settings.title == "New Title"
+
+    @pytest.mark.parametrize("title", [None, "", "t" * (VECTOR_SETTINGS_CREATE_TITLE_MAX_LENGTH + 1)])
+    async def test_update_vector_settings_with_invalid_title(
+        self, async_client: "AsyncClient", owner_auth_header: dict, title: str
+    ):
+        vector_settings = await VectorSettingsFactory.create()
+
+        response = await async_client.patch(
+            f"/api/v1/vectors-settings/{vector_settings.id}",
+            headers=owner_auth_header,
+            json={"title": title},
+        )
+
+        assert response.status_code == 422
+
+    @pytest.mark.parametrize(
+        "payload",
+        [
+            {"title": None},
+        ],
+    )
+    async def test_update_vector_settings_with_invalid_payload(
+        self, async_client: "AsyncClient", owner_auth_header: dict, payload: dict
+    ):
+        vector_settings = await VectorSettingsFactory.create()
+
+        response = await async_client.patch(
+            f"/api/v1/vectors-settings/{vector_settings.id}",
+            headers=owner_auth_header,
+            json=payload,
+        )
+
+        assert response.status_code == 422
```

Comment:
It's ok to be explicit:

```suggestion
    async def test_update_vector_settings_with_title_as_none(
        self, async_client: "AsyncClient", owner_auth_header: dict
    ):
        vector_settings = await VectorSettingsFactory.create()

        response = await async_client.patch(
            f"/api/v1/vectors-settings/{vector_settings.id}",
            headers=owner_auth_header,
            json={"title": None},
        )

        assert response.status_code == 422
```